src/Microsoft.Identity.Abstractions/ApplicationOptions/CredentialDescription.cs

```diff
@@ -15,6 +15,22 @@ namespace Microsoft.Identity.Abstractions
     /// </summary>
     public class CredentialDescription
     {
+        private string? _cachedId;
+
+        /// <summary>
+        /// Gets a unique identifier for a CredentialDescription based on <see cref="SourceType"/> and <see cref="ReferenceOrValue"/>.
+        /// </summary>
+        public string Id
+        {
+            get
+            {
+                if (_cachedId == null)
+                    _cachedId = $"{SourceType}_{ReferenceOrValue}";
```

Comment:
Is it enough? or does it require `$"{SourceType}_{Container}_{ReferenceOrValue}"` ?

Reply:
ReferenceOrValue can be null, Container can also be null. SourceType will not be null. 
Adding Container to the Id will make it stronger. Will update the PR.

Reply:
Update the Id to use Container. 
Today, Id property doesn't throw an exception if Container and/or ReferenceOrValue is null. It proceeds to generate a string that Id.Web will use to maintain a semaphore for that Id.